DOC Final update for CMS5.0 changelog (#255)
silverstripe/developer-docs: en/04_Changelogs/5.0.0.md

- `swiftmailer/swiftmailer` has been removed and replaced with `symfony/mailer`
- Various Symfony dependencies have been upgraded from 4.x to 6.x. A small number of code changes were made in `silverstripe/framework` to work with the symfony 6.x dependencies.
- `silverstripe/graphql` v3 is no longer supported. It is recommended that you ensure you are using `silverstripe/graphl` v4 in your Silverstripe CMS 4 project before upgrading to Silverstripe CMS 5. 
  - Review the [`silverstripe/graphql` v4 upgrade documentation](https://docs.silverstripe.org/en/4/upgrading/upgrading_to_graphql_4/) for more details.
- PHPUnit 5.7 is no longer supported. It is recommended that you ensure your tests are running with PHPUnit 9 in your Silverstripe CMS 4 project before upgrading to Silverstripe CMS 5. 
  - Review the [PHPUnit 9 documentation ](https://docs.silverstripe.org/en/4/upgrading/phpunit9/) for more details.

> - `masterminds/html5` is now used to parse HTML content instead of `DomDocument::loadHTML()`. This may cause slight changes in rendered content if the content being saved contains invalid HTML. The `silverstripe/html5` module is no longer required for rendering HTML 5 content and should be removed.

Reviewer: ```suggestion
- `masterminds/html5` is now used to parse HTML content instead of `DomDocument::loadHTML()`. This may cause slight changes in rendered content if the content being saved contains invalid HTML. The `silverstripe/html5` module is no longer required for rendering HTML 5 content and should be removed from your project.
```

Author: Done

### Webpack config

The [`@silverstripe/webpack-config`](https://www.npmjs.com/package/@silverstripe/webpack-config) and [`@silverstripe/eslint-config`](https://www.npmjs.com/package/@silverstripe/eslint-config) NPM libraries, along with the build stack for all supported modules, have been updated to be compatible with node 18 and webpack 5. This will only impact you if your module or project uses one or both of those NPM packages - you will need to make sure you update your own dependencies to be compatible with them, along with the dependencies listed below if you use them.

`@silverstripe/webpack-config` has a new optional abstracted API to declare the webpack config for transpiling the javascript and css for your CMS customisations.

> Review the [`@silverstripe/webpack-config` read me](https://github.com/silverstripe/webpack-config#readme) to get the list of new APIs.

Reviewer: ```suggestion
Review the [`@silverstripe/webpack-config` read me](https://github.com/silverstripe/webpack-config#readme) to get the list of new APIs and usage documentation.
```
It's not just a list of API

Author: Done

### silverstripe/security-extensions module is folded into core {#security-extensions-is-core}

The functionality from [silverstripe/security-extensions](https://github.com/silverstripe/silverstripe-security-extensions) has been folded into the core modules. You no longer need to install that module to benefit from the features it provided - and indeed there will not be a CMS 5 compatible version of that module. If you have `silverstripe/security-extensions` as a dependency in your composer.json file, you will need to remove it.

With this change you now have access to the following features:

#### Require password change on next log in

Administrators with the ability to administer members can see a checkbox in the CMS under the area to set the member's password. Checking this box will set the password expiry to the current date. The next time the member logs in, they will be required to choose a new password for their account.

> No change is made when setting this field and the password is already expired for auditing purposes (an administrator could see how long ago a password expired). Similarly, no change is made when unsetting this field and the expiry date is in the future, it should remain so - the checkbox is for immediately requiring a new password on the _next_ log in.

Author: > Similarly, no change is made when unsetting this field and the expiry date is in the future, it should remain so - the checkbox is for immediately requiring a new password on the _next_ log in.

@GuySartorelli I don't understand this.

Reviewer: If the date is already in the future, _unchecking_ the box doesn't do anything.

Author: I rewrote this bit so it makes a little more sense to my monkey brain.

- Void HTML elements such as `<img>` are no longer rendered as self-closing tags e.g. `<img src="foo" alt="bar">` is now rendered, previously `<img src="foo" alt="bar" />` was rendered.
- It’s now possible to hook into FormField validation via the `updateValidationResult` extension hook. Module authors releasing form fields should opt-in to this functionality by calling the new `extendValidationResult` method to enable developers to extend and customise validation. See the [form validation documentation](/developer_guides/forms/validation/#extensions) for more information.

## Bugfixes {#bugfixes}

- If a page which is a child of a root-level page gets archived, and then its former parent is removed, it can only be restored if `can_be_root` for that page's class is true.

> This release includes a number of bug fixes to improve a broad range of areas. Check the change logs for full details of these fixes split by module. Thank you to the community members that helped contribute these fixes as part of the release!

Author: This seems like a really random comment to have here. If we want to keep this here, then we should move it to a more sensible place.

Reviewer: ~~The listed bug fix is a breaking behavioural change so it should be here somewhere. Please move it to a more sensible place. Though it is a bug fix, so the bug fix heading seems to be the correct place for it to me.~~ - I now see you were explicitly referring to the bit I mention below.

Also the "this release includes" etc etc - that part is language that appears in _all_ of our changelogs. Is there a reason to exclude it from _this_ changelog?

Author: I think its meant to be an introduction to our full commit list. But there's an API change list in between that an the full list.

I'll move the language down there.

Author: Done

> ## Bugfixes {#bugfixes}

Reviewer: ```suggestion
## Bug fixes {#bug fixes}
```

The automated changelog adds a `#bugfixes` anchor, so we need this one to be unique.
Also update the TOC.

Author: Done

- Changed return type for [`Symbiote\QueuedJobs\Services\QueuedJobHandler::write()`](api:Symbiote\QueuedJobs\Services\QueuedJobHandler::write()) from dynamic to `void`
- Changed parameter type in [`Symbiote\QueuedJobs\Services\QueuedJobHandler::write()`](api:Symbiote\QueuedJobs\Services\QueuedJobHandler::write()) for `$record` from `array` to `Monolog\LogRecord`

### tractorcow/silverstripe-fluent

- Removed deprecated method `TractorCow\Fluent\Extension\FluentExtension::getLinkingMode()` - use `LocaleInformation()` instead
- Removed deprecated method `TractorCow\Fluent\Extension\FluentExtension::LocaleLink()` - use `LocaleInformation` instead

</details>

## Full commits list

> This release includes a number of bug fixes to improve a broad range of areas. Check the change logs for full details of these fixes split by module. Thank you to the community members that helped contribute these fixes as part of the release!

Reviewer: This probably needs to be reworded now that it's not in the bug fixes section - this wording has a heavy emphasis on bug fixes, but the commits list isn't _just_ about bug fixes.

> This release includes a number of bug fixes to improve a broad range of areas. Check the change logs for full details of these fixes split by module. Thank you to the community members that helped contribute these fixes as part of the release!

Author: ```suggestion
This release includes a number of changes to improve a broad range of areas. Check the change logs for full details of these updates split by module. Thank you to the community members that helped contribute these fixes as part of the release!
```

<details>
<summary>Reveal full list of commits</summary>

<!--- Changes below this line will be automatically regenerated -->

<!--- Changes above this line will be automatically regenerated -->

> </details>

Reviewer: Are we likely to do this going forward for future changelogs?
Your answer doesn't have any bearing on this PR, I'm just curious.

Reviewer: Note also that there's a `changelog` heading that gets put in there automatically that we'll want to remove when it does get generated, since you've added an equivalent heading above this.

Author: I don't think this is normally this problematic for minor release changelogs because they tend to be smaller and distinction between the "Release Notes" part and the commit logs is clearer.

Still, if it works well, then maybe we want to adopt this approach going forward.
